res/res_agi.c

```diff
@@ -1091,111 +1090,85 @@ static int add_silence(struct ast_channel *chan, struct ast_frame *f, struct ast
         long int ts_diff=0, f_ptime, last_seq=0;
 	int64_t gap_ms=0;
 	struct timeval s_tv = ast_channel_get_rec_start_time(chan);
+	unsigned int themssrc=0;
+	int ssrc_change=0;
+
+        /*! Check SSRC */
+        if ((f->themssrc != 0) && (f->themssrc != ast_channel_get_last_ssrc(chan, stream_no))){
+                ast_debug(1, "STREAM %d ==== SSRC changed (%u) to (%u)\n", stream_no, ast_channel_get_last_ssrc(chan, stream_no), f->themssrc);
+                ast_channel_set_last_ssrc(chan, f->themssrc, stream_no);
+                themssrc = f->themssrc;
+		ssrc_change=1;
+        } else if (f->themssrc == ast_channel_get_last_ssrc(chan, stream_no)) {
+                themssrc = ast_channel_get_last_ssrc(chan, stream_no);
+        }else {
+		//Discard the packet
```

Comment:
please add detailed comment on why the packet is discarded so that we can recall later.